### DISCUSSION TITLE
Moving Multirotor UAV right and left w.r.t to its position

### DISCUSSION THREAD

**Discussion by seandiacono** [microsoft/AirSim#3641] [Support Q&A] [answered]
This may be a trivial question however I can't seem to find a simple solution yet. I am creating an obstacle avoidance system for the Multirotor client and I would like the drone to move to the left or right depending on where the object is in its FoV. 

My issue is moving the drone to the left or right with respect to its position. I am using the SimpleFlight controller and I can't find a command which will consistently move the drone to the left or the right no matter the orientation of the drone. 

I've tried using moveByVelocityAsync, naturally, this doesn't work since the x and y coordinates are global rather than for the drone. I also tried to use moveByMotorPWMsAsync although I'm not entirely sure if I understand how it works so I did not manage to use it. The 1.2.8 API docs also state that there is an API call moveByVelocityBodyFrameAsync which moves the drone with respect to the vehicle's local NED frame however when trying to use it an error is thrown stating that the command does not exist.

Is there some simple solution that I am missing or is this problem not as simple as I think it is?

**Comment by rajat2004** [comment, marked as answer]
I think there are 2 questions here. First, which version of AirSim simulator are you using, i.e is it source built from master branch, or downloaded from [Releases](https://github.com/microsoft/AirSim/releases)? If latest master, then great, else you'll have to use the 1.4.0 binaries.

Next is the version of AirSim Python API. The complicated part here is that there's no Python package release corresponding to 1.4.0, the [PyPI package](https://pypi.org/project/airsim/) is 1.3.0 are older than 1.4.0 release (being discussed on https://github.com/microsoft/AirSim/issues/3634). PR https://github.com/microsoft/AirSim/pull/3169 added the API, which is later than the last release. 
Therefore, you should use the latest master Python API. I can tell you about 2 ways for this -
1. Use [setup_path.py](https://github.com/microsoft/AirSim/blob/master/PythonClient/multirotor/setup_path.py). Place your files in one of the folders inside `PythonClient` like `multirotor`, `car`, etc. (or create one to keep things separate, and copy the `setup_path.py` file from another folder). Add `import setup_path` before `import airsim` in your files. Now the latest master API (or any branch currently checked out) will be used.
2. Use an editable pip install i.e `pip install -e .` from inside the `PythonClient` folder. It'll automatically use the currently checked-out branch files for this.

I would strongly suggest using a virtual env for doing any Python package stuff, otherwise it can get messy very easily and you can break stuff. Also, the [API Ref](https://microsoft.github.io/AirSim/api_docs/html/) follows the master and not the release

So finally, use the latest binaries or source build AirSim, and the latest Python API and it should work.

Should add this info to the docs as well I think (updated in https://github.com/microsoft/AirSim/pull/3611)

**Reply by seandiacono** [reply]
I downloaded the latest 1.4.0 release and updated the Python API to the latest version and this worked. Thanks!
